CommunityCenterOverhaul/Framework/ImageEditor.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Linq;
+using Microsoft.Xna.Framework.Graphics;
+using StardewConfigFramework;
+using StardewModdingAPI;
+
+namespace CommunityCenterBundleOverhaul.Framework
+{
+    internal class ImageEditor : IAssetEditor
+    {
+        /*********
+        ** Properties
+        *********/
+        private readonly IModHelper Helper;
+        private readonly IMonitor Monitor;
+        private readonly ModOptionSelection DropDown;
+
+
+        /*********
+        ** Public methods
+        *********/
+        public ImageEditor(IModHelper helper, IMonitor monitor, ModOptionSelection dropDown)
+        {
+            this.Monitor = monitor;
+            this.Helper = helper;
+            this.DropDown = dropDown;
+        }
+
+        public bool CanEdit<T>(IAssetInfo asset)
+        {
+            return this.DropDown.SelectionIndex != 0 && asset.AssetNameEquals(@"LooseSprites\JunimoNote");
+        }
+
+        public void Edit<T>(IAssetData asset)
+        {
+            this.GetBundleTexturefromJson(this.DropDown.SelectionIndex, asset, this.Helper.Translation.Locale);
+        }
+
+
+        /*********
+        ** Private methods
+        *********/
+        private void GetBundleTexturefromJson(int selInd, IAssetData asset, string locale)
+        {
+            // get bundle
+            Bundle[] data = this.Helper.ReadJsonFile<Bundle[]>(@"bundles\bundles.json");
+            Bundle bundle = data.FirstOrDefault(p => p.ID == selInd);
+            if (bundle == null)
+                return;
+
+            // edit asset
+            switch (selInd)
+            {
+                case 0:
+                    {
+                        string filenameLocale = this.GetFilenameLocale(locale);
+                        string filename = filenameLocale != null
+                            ? $@"LooseSprites\\JunimoNote.{filenameLocale}.xnb"
+                            : @"LooseSprites\\JunimoNote.xnb";
+                        Texture2D texture = this.Helper.Content.Load<Texture2D>(filename, ContentSource.GameContent);
+                        asset.AsImage().PatchImage(texture);
+                    }
+                    break;
+
+                case 1:
+                case 2:
```

Comment:
Case 2 and Case 3 must have a different Texture. Look at bundles\images. 2 and 3 are for bundle version v0.2a and v0.2b 1 and 4 for Bundle version v0.2 and v0.2c (I should maybe remove images which are then there twice)

Reply:
See original source: https://github.com/chaos234/StardewValleyMods/blob/master/CommunityCenterOverhaul/CCBOJunimoEditor.cs#L83-L96

Reply:
The cases do exactly the same thing in the original code, though.

Reply:
https://github.com/chaos234/StardewValleyMods/blob/master/CommunityCenterOverhaul/CCBOJunimoEditor.cs#L48-L56 < THere is de defination for var bundleName :)

Reply:
That logic is still there (the method was renamed to `GetBundleTextureFromJson`). Since the refactor matches your previous logic, I suggest accepting it and then making your further changes. :)

Reply:
Ah I see, then I have said nothing ^^